sdk/identity/identity/test/manual-integration/AzureFunctions/arm-template.json

```diff
@@ -0,0 +1,236 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "vaults_name": {
+      "defaultValue": "kaghiya-standard-kv",
+      "type": "string"
+    },
+    "appName": {
+      "type": "string",
+      "defaultValue": "[concat('fnapp', uniqueString(resourceGroup().id))]",
+      "metadata": {
+        "description": "The name of the function app that you wish to create."
+      }
+    },
+    "storageAccountType": {
+      "type": "string",
+      "defaultValue": "Standard_LRS",
+      "allowedValues": ["Standard_LRS", "Standard_GRS", "Standard_RAGRS"],
+      "metadata": {
+        "description": "Storage Account type"
+      }
+    },
+    "location": {
+      "type": "string",
+      "defaultValue": "[resourceGroup().location]",
+      "metadata": {
+        "description": "Location for all resources."
+      }
+    },
+    "appInsightsLocation": {
+      "type": "string",
+      "metadata": {
+        "description": "Location for Application Insights"
+      }
+    },
+    "runtime": {
+      "type": "string",
+      "defaultValue": "node",
+      "allowedValues": ["node", "dotnet", "java"],
+      "metadata": {
+        "description": "The language worker runtime to load in the function app."
+      }
+    }
+  },
+  "variables": {
+    "vaultname": "[parameters('vaults_name')]",
+    "functionAppName": "[parameters('appName')]",
+    "hostingPlanName": "[parameters('appName')]",
+    "applicationInsightsName": "[parameters('appName')]",
+    "storageAccountName": "[concat(uniquestring(resourceGroup().id), 'azfunctions')]",
+    "functionWorkerRuntime": "[parameters('runtime')]"
+  },
+  "resources": [
+    {
+      "type": "Microsoft.KeyVault/vaults",
+      "apiVersion": "2020-04-01-preview",
+      "name": "[variables('vaultname')]",
+      "location": "westus2",
+      "properties": {
+        "sku": {
+          "family": "A",
+          "name": "Standard"
+        },
+        "tenantId": "72f988bf-86f1-41af-91ab-2d7cd011db47",
+        "accessPolicies": [
+          {
+            "tenantId": "72f988bf-86f1-41af-91ab-2d7cd011db47",
+            "objectId": "612e5e44-2c95-4bd0-96b7-3db4dae3b560",
```

Comment:
This won't work because it is hardcoded to your user. Two options:

1. We can remove the access policies section entirely since we are providing instructions on how to set up the access policies
2. We can make the access policies smarter by dynamically setting them to the objectId of the function app. 

Happy to help with either one of course. (1) will be easier for us, so maybe that's good enough?

Reply:
If we do keep this here, we only need secrets permission so we can delete keys and certs

Reply:
If we decided to go with (2) I believe we can set this up easily using:

```
[reference(resourceId('Microsoft.Web/sites', variables('functionAppName')), '2019-08-01', 'full').identity.principalId]
``` 

once you enable system assigned managed identity in the arm template (see my other comment about that).

Reply:
Here's a complete example, which actually does exactly what we want 😄 

https://www.codeisahighway.com/there-is-a-new-way-to-reference-managed-identity-in-arm-template/